launch/config.py

```diff
@@ -0,0 +1,385 @@
+    config_dir to make an absolute path
+    """
+    if path.startswith('/'):
+        # nothing to do, return the path
+        return path
+    if path.startswith('~'):
+        # Try this and raise LauncherError
+        home_dir = os.environ['HOME']
+        return path.join(home_dir, path[1:])
+    # path is relative to config_dir
+    return os.path.join(config_dir, path)
+
+
+def get_validated_config(config_path: str):
+    """ Takes a dcos-launch configuration file and:
+    - string-ifies nested dicts
+    - expands paths into absolute paths, supporting tilde expansion
+    - converts provider_info_filename into provider_info
+    """
+    config_dir = os.path.dirname(os.path.abspath(config_path))
+    # Try and catch YamlParseError
+    config = load_yaml(config_path)
+    for k, v in config.items():
+        if k.endswith('_filename'):
+            config[k] = expand_path(v, config_dir)
+        if k == 'dcos_config':
+            for kk, vv in config[k].items():
+                if kk.endswith('_filename'):
+                    config[k][kk] = expand_path(vv, config_dir)
+        if isinstance(config[k], dict):
+            config[k] = yaml.dump(config[k])
+    config = stringify_configuration(config)
+    return validate_config(config)
+
+
+def get_user_argument_source(user_config: dict):
+    """ Unlike gen, nested config are allows here
+    """
+    user_source = Source(is_user=True)
+    for name, value in user_config.items():
+        user_source.add_must(name, value)
+    return user_source
+
+
+def validate_config(user_config: dict):
+    user_source = get_user_argument_source(user_config)
+    t = get_target()
+    sources = [get_source(), user_source]
+    resolver = resolve_configuration(sources, [t])
+    status = resolver.status_dict
+
+    if status['status'] == 'errors':
+        raise launch.util.LauncherError('ValidationError', pretty_print_validate_error(status))
+
+    late_variables = dict()
+
+    for source in sources:
+        for setter_list in source.setters.values():
+            for setter in setter_list:
+                if not setter.is_late:
+                    continue
+
+                if setter.name not in resolver.late:
+                    continue
+
+                # Skip late vars that aren't referenced by config.
+                if not resolver.arguments[setter.name].is_finalized:
+                    continue
+
+                # Validate a late variable should only have one source.
+                assert setter.name not in late_variables
+
+                late_variables[setter.name] = setter.late_expression
+
+    final_args = {k: v.value for k, v in resolver.arguments.items() if v.is_finalized}
+    unrecognized_args = set(user_config.keys()) - set(final_args.keys())
+    if len(unrecognized_args) > 0:
+        raise launch.util.LauncherError(
+            'ValidationError', 'Unrecognized/incompatible arguments: {}'.format(unrecognized_args))
```

Comment:
Sounds like that'll take some work, bummer. Yeah if you could file a ticket and link it here in a comment, that'll help surface the problem.